src/node_buffer.cc

```diff
@@ -786,14 +787,29 @@ void StringWrite(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(written);
 }
 
-void ByteLengthUtf8(const FunctionCallbackInfo<Value> &args) {
+void SlowByteLengthUtf8(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
   CHECK(args[0]->IsString());
 
   // Fast case: avoid StringBytes on UTF8 string. Jump to v8.
   args.GetReturnValue().Set(args[0].As<String>()->Utf8Length(env->isolate()));
 }
 
+uint32_t FastByteLengthUtf8(Local<Value> receiver,
+                            const v8::FastOneByteString& source) {
+  uint32_t result = 0;
+  uint32_t length = source.length;
+  const uint8_t* data = (const uint8_t*)(source.data);
+  for (uint32_t i = 0; i < length; ++i) {
+    result += (data[i] >> 7);
+  }
```

Comment:
@anonrig I don't know about SIMD or GCC/Clang auto-vectorization in Node, but this should be relatively trivial to vectorize?

Reply:
I had chatted about this with @anonrig a while ago – yes, it is: https://gist.github.com/addaleax/601f5380a53257b80f373ba53376181f

Reply:
Could we include that?

Reply:
I think we can switch to something implemented in simdutf later when they have it (they already have plans to implement latin1 to UTF8 methods https://github.com/nodejs/performance/issues/52#issuecomment-1427128144), though for the purpose of this patch (moving it to fast calls) what it has now is probably good enough.

Reply:
@ronag In addition to what @joyeecheung said, I’m not really up for packing that into Node.js & making sure it passes CI and works everywhere – simdutf already has all the logic for dispatching different versions of an algorithm based on CPU features etc., so that is probably a nicer place for that code to live